components/tenant-fetcher/internal/authenticator/authenticator_test.go

```diff
@@ -0,0 +1,5 @@
+package authenticator
+
+func (a *Authenticator) SetJWKSEndpoint(url string) {
```

Comment:
Maybe you can move this function to the middaware_test.go file and delete this file?

Reply:
Addressed it. 🚀